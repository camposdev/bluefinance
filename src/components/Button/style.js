import styled from "styled-components";

export const ButtonComponent = styled.button`
  position: relative;
  border: none;
  padding: 5px ${({ large }) => large ? "40px" : "20px"};
  height: ${({ large }) => large ? "4.8rem" : "3.4rem"};
  text-transform: uppercase;
  letter-spacing: 0.2rem;
  font-weight: bold;
  border-radius: 2.4rem;
  font-size: ${({ large }) => large ? "1.4rem" : "1.2rem"};
  cursor: pointer;
  overflow: hidden;
  transition: all 0.3s ease;

  ${({ light }) => light ? `
    color: var(--primary);
    background: var(--gradient-light);
    box-shadow: 0 3px 10px rgba(255, 255, 255, 0.3);

    &::before {
      background: var(--gradient-primary);
    }

    &:hover {
      box-shadow: 0 8px 20px rgba(255, 255, 255, 0.3);
    }
    &:active {
      box-shadow: 0 0 0 rgba(255, 255, 255, 0.3);
    }
  ` : `
    color: white;
    background: var(--gradient-primary);
    box-shadow: 0 3px 10px rgba(44, 106, 255, 0.3);

    &::before {
      background: var(--gradient-light);
    }

    &:hover {
      box-shadow: 0 8px 20px rgba(44, 106, 255, 0.3);
    }
    &:active {
      box-shadow: 0 0 0 rgba(44, 106, 255, 0.3);
    }
  `}

  &::before {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    opacity: 0;
    transition: opacity 0.3s ease;
  }

  &:hover {
    &::before {
      opacity: 0.2;
    }
  }
`;