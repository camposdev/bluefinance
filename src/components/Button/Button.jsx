import React from "react";
import PropTypes from "prop-types";
import { ButtonComponent } from "./style";


const Button = ({ children, ...props }) => <ButtonComponent {...props}>{children}</ButtonComponent>

Button.propTypes = {
  children: PropTypes.any.isRequired,
  light: PropTypes.bool,
  large: PropTypes.bool
};

Button.defautProps = {
  light: false,
  large: false
};

export default Button;