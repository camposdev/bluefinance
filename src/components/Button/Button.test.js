import React from "react";
import renderer from "react-test-renderer";
import Button from "./Button";

describe("Button component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Button>Click me!</Button>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
