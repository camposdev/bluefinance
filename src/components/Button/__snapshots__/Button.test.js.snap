// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<button
  className="sc-bdVaJa fsBEfH"
>
  Click me!
</button>
`;
